src/app/pages/home-search/home-search.page.ts

```diff
@@ -1,121 +1,248 @@
+          }
+        }
+        this.extractLabels(dataReturned.data.data.selectedFilters);
+        this.getFilteredData();
+      }
+      this.page = 1;
+      this.setPaginatorToFirstpage = true;
+      this.fetchSessionList()
+    });
+    modal.present()
   }
-  cancelSearch(event: any){
-    this.searching = true
-    this.results = []
+
+  async fetchSessionList() {
+    var obj={page: this.page, limit: this.limit, type: this.type, searchText : this.searchText, selectedChip : this.criteriaChipName, filterData : this.urlFilterData}
+    var response = await this.sessionService.getSessionsList(obj);
+    this.results = response?.result?.data;
+    this.totalCount = response.result.count;
+    this.noDataMessage = obj.searchText ? "SEARCH_RESULT_NOT_FOUND" : "THIS_SPACE_LOOKS_EMPTY"
   }
-  trimLeft(inputString: string): string {
-    return this.searchText = inputString.replace(/^\s+/, '');
+
+  onPageChange(event){
+    this.page = event.page,
+    this.pageSize = event.pageSize;
+    this.fetchSessionList()
   }
 
-  search(){
-    if(this.searchText!=""){
-      switch(this.type) {
-        case 'all-sessions':
-          this.getSessionsList();
+  async eventAction(event) {
+    this.user = await this.localStorage.getLocalData(localKeys.USER_DETAILS)
+    if (this.user.about) {
+      switch (event.type) {
+        case 'cardSelect':
+          this.router.navigate([`/${CommonRoutes.SESSIONS_DETAILS}/${event.data.id}`]);
           break;
-        case 'mentor-profile':
-          this.getMentorList();
+
+        case 'joinAction':
+          await this.sessionService.joinSession(event.data)
+          this.fetchSessionList()
+          break;
+
+        case 'enrollAction':
+          let enrollResult = await this.sessionService.enrollSession(event.data.id);
+          if (enrollResult.result) {
+            this.toast.showToast(enrollResult.message, "success")
+            this.fetchSessionList()
+          }
+          break;
+
+        case 'startAction':
+          this.sessionService.startSession(event.data.id).then(async () => {
+            var obj = { page: this.page, limit: this.limit, searchText: "" };
+            if(this.profileService.isMentor){
+              this.createdSessions = await this.sessionService.getAllSessionsAPI(obj);
+            }
+          })
           break;
       }
     } else {
-      this.toast.showToast("Please provide a valid text for search","danger")
+      this.profileService.upDateProfilePopup()
     }
   }
 
-  segmentChanged(event){
-    this.results=[]
-    this.type = event.target.value;
-    if(this.searchText !== "")
-    this.search();
+  locationBack(){
+    this.location.back()
   }
 
-  async onSessionAction(event){
-    switch (event.type) {
-      case 'cardSelect':
-        this.router.navigate([`/${CommonRoutes.SESSIONS_DETAILS}/${event.data.id}`])
-        break;
+  extractLabels(data) {
```

Comment:
do we need this? from filter itself u can get label and name right? why extra function